c10/core/Event.h

```diff
@@ -0,0 +1,145 @@
+#pragma once
+
+#include "ATen/Context.h"
+#include "c10/util/Optional.h"
+#include "c10/util/Exception.h"
+#include "c10/core/Stream.h"
+
+#include <memory>
+#include <limits>
+#include <atomic>
+#include <utility>
+
+namespace at { namespace cuda {  namespace detail {
+    struct CUDAHooks;
+}}} // at::cuda::detail
+
+namespace c10 {
+
+/*
+* A generic Event capable of representing device-specific events when the appropriate
+* backend is loaded. It is designed to dispatch calls to different backends
+* based on the Stream the event is recorded on.
+*
+* This class is not thread-safe and is movable but not copyable. 
+*
+*/
+struct Event final {
+
+static constexpr unsigned int DEFAULT_FLAGS = std::numeric_limits<unsigned int>::max();
+friend class at::cuda::detail::CUDAHooks;
+
+// Constructor optionally accepts flags.
+// Note: DEFAULT_FLAGS reflects PyTorch's defaults, not the default values
+// for constructing a particular backend's events. For example, CUDA events
+// are constructed by PyTorch with the cudaEventDisableTiming flag, which is
+// not the default constructor behavior for a CUDA event. To obtain a CUDA event
+// with the CUDA default flags pass zero (=cudaEventDefault) explicitly. 
+Event(const unsigned int flags = DEFAULT_FLAGS) 
+: flags_{flags}
+{ }
+
+// Copy constructor and copy assignment operator (deleted)
+Event(const Event&) = delete;
+Event& operator=(const Event&) = delete;
+
+
+// Move constructor and move assignment operator (std::swap)
+Event(Event&& other) { moveHelper(std::move(other)); }
```

Comment:
To future proof, this should call the no-args constructor.

Reply:
Made it call a constructor using properties of other.